File: backend/models/domain/Summary.py
from flask_restful import fields


class Summary():
    resource_fields = {
        'amountStart': fields.Float,
        'amountEnd': fields.Float,
        'totalCredit': fields.Float,
        'totalDebit': fields.Float,
        'periodType': fields.String
    }

    def __init__(self, amountStart=None, amountEnd=None, totalCredit=None, totalDebit=None, periodType=None) -> None:
        self.amountStart = amountStart
        self.amountEnd = amountEnd
        self.totalCredit = totalCredit
        self.totalDebit = totalDebit
        self.periodType = periodType

    def __repr__(self):
        return '<Summary %r>' % self.periodType

File: frontend/src/app/modules/statistics/summary.ts
export class Summary {

  public amountStart: number;
  public amountEnd: number;
  public totalCredit: number;
  public totalDebit: number;
  public periodType: 'DAY'|'MONTH'|'YEAR';

  constructor(amountStart: number, amountEnd: number, totalCredit: number, totalDebit: number, periodType) {
    this.amountStart = amountStart;
    this.amountEnd = amountEnd;
    this.totalCredit = totalCredit;
    this.totalDebit = totalDebit;
    this.periodType = periodType;
  }
}
